consensus/hotstuff/signature/packer.go

```diff
@@ -158,88 +158,86 @@ func (p *ConsensusSigDataPacker) Unpack(blockID flow.Identifier, signerIDs []flo
 		RandomBeaconSigners:          randomBeaconSigners,
 		AggregatedStakingSig:         data.AggregatedStakingSig,
 		AggregatedRandomBeaconSig:    data.AggregatedRandomBeaconSig,
-		ReconstructedRandomBeaconSig: data.RandomBeacon,
+		ReconstructedRandomBeaconSig: data.ReconstructedRandomBeaconSig,
 	}, nil
 }
 
 // the total number of bytes required to fit the `count` number of bits
 func bytesCount(count int) int {
-	totalBytes := count / 8
-	if count%8 > 0 {
-		totalBytes++
-	}
-	return totalBytes
+	return (count + 7) >> 3
 }
 
-// serializeToBytes encodes the given sigTypes into a bit vector.
+// serializeToBitVector encodes the given sigTypes into a bit vector.
 // We append tailing `0`s to the vector to represent it as bytes.
-func serializeToBytes(sigTypes []hotstuff.SigType) ([]byte, error) {
+func serializeToBitVector(sigTypes []hotstuff.SigType) ([]byte, error) {
 	totalBytes := bytesCount(len(sigTypes))
 	bytes := make([]byte, 0, totalBytes)
-	// a sig type can be converted into one bit.
-	// so every 8 sig types will fit into one byte.
-	// iterate through every 8 sig types, for each sig type in each
-	// group, convert into each bit, and fill the bit into the byte.
+	// a sig type can be converted into one bit, the type at index 0 being converted into the least significant bit:
+	// the random beacon type is mapped to 1, while the staking type is mapped to 0.
 	// the remaining unfilled bits in the last byte will be 0
-	for byt := 0; byt < len(sigTypes); byt += 8 {
-		b := byte(0)
-		offset := 7
-		for pos := byt; pos < byt+8 && pos < len(sigTypes); pos++ {
-			sigType := sigTypes[pos]
-
-			if !sigType.Valid() {
-				return nil, fmt.Errorf("invalid sig type: %v at pos %v", sigType, pos)
-			}
-
-			b ^= byte(sigType) << offset
```

Comment:
This line rely on `sigType` constants being `1` or `0` while there is no guarantee for it.